src/realm/table.cpp

```diff
@@ -5956,9 +5956,19 @@ void Table::generate_patch(const Table* table, std::unique_ptr<HandoverPatch>& p
     if (table) {
         patch.reset(new Table::HandoverPatch);
         patch->m_table_num = table->get_index_in_group();
-        // must be group level table!
-        if (patch->m_table_num == npos) {
-            throw std::runtime_error("Table handover failed: not a group level table");
+        patch->m_is_sub_table = (patch->m_table_num == npos);
+
+        if (patch->m_is_sub_table) {
+            auto col = dynamic_cast<SubtableColumn*>(table->m_columns.get_parent());
+            if (col) {
+                Table* parent_table = col->m_table;
```

Comment:
What about subtables of subtables? Don't you need to generate patches recursively until you hit a group level table?